flake.nix

```diff
@@ -2,71 +2,95 @@
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
     rust-overlay.url = "github:oxalica/rust-overlay";
+    flake-parts.url = "github:hercules-ci/flake-parts";
```

Comment:
Why do we need `flake-parts`?

Reply:
Main reason is, that it's a more elegant way to avoid [forAllSystems](https://github.com/eneoli/flakeshot/blob/main/flake.nix#L9-L17) and doing that repeatedly for most entries, like `apps`, `packages`, etc.

So instead of
```nix
apps = forAllSystems (pkgs: ...);
packages = forAllSystems (pkgs: ...);
```
we can do
```nix
apps = <definition for `apps`>
packages = <definition for `packages`>
```